File: create-fusee-go/internal/assets/embed/framework/engine-go/load-helpers.go
package engine

import "fmt"

type RouteRedirect struct {
	Status   int
	Location string
}

func (e *RouteRedirect) Error() string {
	return fmt.Sprintf("redirect(%d) to %s", e.Status, e.Location)
}

type RouteHTTPError struct {
	Status  int
	Message string
}

func (e *RouteHTTPError) Error() string {
	return e.Message
}

func Redirect(status int, location string) error {
	return &RouteRedirect{Status: status, Location: location}
}

func HTTPError(status int, message string) error {
	return &RouteHTTPError{Status: status, Message: message}
}

File: framework/server-legacy/load-helpers.js
export class RouteRedirect extends Error {
    /** Creates a control-flow error representing an HTTP redirect response. */
    constructor(status, location) {
        super(`redirect(${status}) to ${location}`)
        this.name = 'RouteRedirect'
        this.status = status
        this.location = location
    }
}

export class RouteHttpError extends Error {
    /** Creates a control-flow error representing an intentional HTTP failure. */
    constructor(status, message) {
        super(message)
        this.name = 'RouteHttpError'
        this.status = status
    }
}

/** Aborts route loading and asks the dispatcher to send a redirect. */
export function redirect(status, location) {
    throw new RouteRedirect(status, location)
}

/** Aborts route loading and asks the dispatcher to send an HTTP error. */
export function httpError(status, message) {
    throw new RouteHttpError(status, message)
}